### PR TITLE
introducing integrity

### PR DESCRIPTION
**Pull request by dparalen**
To test:
```
sudo -u apache pulp-integrity --check size --check broken_rpm_symlinks | jq '[.report[].repository] | unique'
```

TODO:
- review/refactor the validator module to rely more on mixins than inheritance

### CODE REVIEW

**Comment by asmacdo** [comment]
Is there an issue associated with this? Specifically, I'd like to see a summary of the motivation for this change.

**Comment by dparalen** [comment]
@asmacdo https://pulp.plan.io/issues/2619 I forgotten a link in the commit message (just #ID there), will submit with next update

File: pulp_integrity/pulp_integrity/generic.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            return validator.ValidationFailure(self, unit, repository,
+                                               BrokenSymlink(unit, repository, link))
+
+    def validate(self, unit, *args):
+        check_func = (
+            unit.downloaded and
+            functools.partial(os.path.samefile, unit.storage_path) or
+            os.path.lexists
+        )
+        symlink_name = unit.get_symlink_name()
```
Comment (dkliban) [inline pulp_integrity/pulp_integrity/generic.py]: This method is not available in older versions of pulp_rpm.

File: pulp_integrity/pulp_integrity/generic.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+            # double remove might happen if same unit is encountered twice
+            raise MissingStoragePath(unit, exc=exc)
+
+    @property
+    def results(self):
+        for storage_path in self.paths:
+            yield validator.DarkPath(self, storage_path, DarkContentError(storage_path))
+
+
+class BrokenSymlinksValidator(validator.MultiValidator):
```
Comment (dkliban) [inline pulp_integrity/pulp_integrity/generic.py]: It looks like this validator is very specific to pulp_rpm.

**Comment by dparalen** [comment]
@dkliban, thanks for the review!
moving the rpm tools to the pulp_rpm repo; see https://github.com/pulp/pulp_rpm/pull/1104

**Pull request closed without merging**